### PR TITLE
[Kernel] Fix awq error when n is not divisable by 128

### PR DESCRIPTION
**Pull request by jinzhen-lin**
The result of vllm awq kernel would be wrong when n is not divisable by 128 (but divisable by 64).

test data

```python
import torch
import safetensors.torch
import vllm._custom_ops as vllm_ops
import awq_ext

k = 128
n = 64
group_size = 64
torch.cuda.manual_seed(0)
qweight = torch.randint(16, (k, n), dtype=torch.uint8, device="cuda")
qweight = qweight[:, 1::2] * 16 + qweight[:, ::2]
qweight = qweight.view(torch.int32)

qzeros = torch.randint(16, (k // group_size, n), dtype=torch.uint8, device="cuda")
qzeros = qzeros[:, 1::2] * 16 + qzeros[:, ::2]
qzeros = qzeros.view(torch.int32)

scales = torch.randn((k // group_size, n), dtype=torch.half, device="cuda")

inputs = torch.eye(qweight.size(0), dtype=torch.half, device="cuda")
```

true result (awq_ext is from https://github.com/casper-hansen/AutoAWQ_kernels):
```
print(`awq_ext`.gemm_forward_cuda(inputs, qweight, scales, qzeros, 8))

tensor([[  2.5449,   0.7163,   6.4141,  ...,   6.4102,   1.9980,  10.7031],
        [ 12.7266,  -3.5820,   1.9736,  ...,   3.2051,   1.4648,  -2.1406],
        [  2.5449,   0.7163,   5.9219,  ...,   8.0156,   1.8643,  -2.1406],
        ...,
        [  2.2324,   0.0000, -11.0000,  ...,   1.2578,  -8.2812,   0.6060],
        [-12.2812,   2.5176,   3.6680,  ...,   2.5156,  -9.6641,   0.0000],
        [-11.1641,   1.6787,   3.6680,  ...,   2.1562, -15.1875,  -0.3030]],
       device='cuda:0', dtype=torch.float16)

```

main (wrong result):
```
print(vllm_ops.gemm_forward_cuda(inputs, qweight, scales, qzeros, 8))


tensor([[2.5449, 0.7163, 6.4141,  ..., 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
        ...,
        [0.0000, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000]],
       device='cuda:0', dtype=torch.float16)
```

this pr (true result):
```
print(vllm_ops.gemm_forward_cuda(inputs, qweight, scales, qzeros, 8))

tensor([[  2.5449,   0.7163,   6.4141,  ...,   6.4102,   1.9980,  10.7031],
        [ 12.7266,  -3.5820,   1.9736,  ...,   3.2051,   1.4648,  -2.1406],
        [  2.5449,   0.7163,   5.9219,  ...,   8.0156,   1.8643,  -2.1406],
        ...,
        [  2.2324,   0.0000, -11.0000,  ...,   1.2578,  -8.2812,   0.6060],
        [-12.2812,   2.5176,   3.6680,  ...,   2.5156,  -9.6641,   0.0000],
        [-11.1641,   1.6787,   3.6680,  ...,   2.1562, -15.1875,  -0.3030]],
       device='cuda:0', dtype=torch.float16)
```

### CODE REVIEW

File: csrc/quantization/awq/gemm_kernels.cu [hunk 1]
```diff
@@ -334,7 +334,7 @@ __global__ void __launch_bounds__(64)
   }
 
   // TODO: Shang: Hoist loop invariance.
-  for (int ax1_0_1 = 0; ax1_0_1 < 4; ++ax1_0_1) {
+  for (int ax1_0_1 = 0; ax1_0_1 < (N / 32); ++ax1_0_1) {
```
Comment (mgoin) [inline csrc/quantization/awq/gemm_kernels.cu:337]: Should we add an assert in awq.py that the N dim is divisible by 32?

**Review by mgoin — APPROVED** [review]
Thank you for figuring out the root issue! This is clearly an issue

File: csrc/quantization/awq/gemm_kernels.cu (hunk 1 above)
Comment (jinzhen-lin) [reply csrc/quantization/awq/gemm_kernels.cu:337]: N is the template parameter, not the output dimension of weight.
